Fix update to latest pydantic

The latest pydantic version includes:
```
 E   pydantic.warnings.PydanticDeprecatedSince212: Using `@model_validator` with mode='after' on a classmethod is deprecated. Instead, use an instance method. See the documentation at https://docs.pydantic.dev/2.12/concepts/validators/#model-after-validator. Deprecated in Pydantic V2.12 to be removed in V3.0.
``` 
we should fix this by, in order of preference:
* Updating our code
* Ignoring the deprecation and spinning off a new issue
* Pinning and spinning off a new issue

## Acceptance Criteria
- CI successfully runs

--- a/src/mx_bluesky/hyperion/parameters/rotation.py
+++ b/src/mx_bluesky/hyperion/parameters/rotation.py
@@ -151,7 +151,6 @@ class RotationScan(RotationExperiment, SplitScan):
 
     @model_validator(mode="after")
     def correct_start_vds(self) -> Any:
-        # assert isinstance(values, RotationScan)
         start_img = 0.0
         for scan in self.rotation_scans:
             scan.nexus_vds_start_img = int(start_img)

